### PR TITLE
Delay initialization (lazy loading)

### PR DESCRIPTION
**Pull request by silesky**
```ts
import { AnalyticsBrowser } from '@segment/analytics-next'

const analytics = new AnalyticsBrowser() // nothing loaded yet

// queue events
analytics.track('foo')
analytics.identify('bar')

if (userConsentsToTracking) {
  analytics.load({ writeKey: 'foo' }) // loaded. queued events should now be sent.
}


```

### CODE REVIEW

**Comment by zikaari** [comment]
I'm curious wouldn't this have worked anyway? We've always buffered up events until `load`, and flushed them when loaded. Can you explain behavioral differences between the existing implementation and this one?

File: packages/browser/src/browser/index.ts [hunk 1]
```diff
@@ -309,12 +309,46 @@ async function loadAnalytics(
...
+        loadAnalytics(settings, options, buffer)
+      )
+    )
+
+    this._resolveLoadStart = (settings, options) =>
+      resolveLoadStart([settings, options])
+  }
+
+  /**
+   * Starts loading an analytics instance including:
```
Comment (chrisradek) [inline packages/browser/src/browser/index.ts:340]: Can you add a note that this should only be called once? Want to make it clear that calling this with different settings won't essentially 'reload' analytics.

File: packages/browser/src/lib/create-deferred.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+/**
+ * Return a promise that can be externally resolved
+ */
+export const createDeferred = <T>() => {
```
Comment (chrisradek) [inline packages/browser/src/lib/create-deferred.ts:4]: This is a better name than `extractPromiseParts` :grin:

File: README.md [hunk 3]
```diff
@@ -53,6 +53,29 @@ document.body?.addEventListener('click', () => {
 })
 ```
 
+## Lazy / Delayed Loading
```
Comment (chrisradek) [inline README.md:56]: Thanks for adding this section!

**Review by chrisradek — APPROVED** [review]

**Pull request merged**